### PR TITLE
feat: add POST /upload endpoint

### PR DESCRIPTION
**Pull request by Sekhmet**
Closes: https://github.com/snapshot-labs/pineapple/issues/28

- Adds new endpoint for uploading images. Input images has to be under 1MB, images are cropped to 500x500 and stored as webp.
- Updates providers to support uploading raw files.

## Test plan
- Run server
- Do `curl -F "file=@./test/unit/providers/fixtures/sample.webp" http://localhost:3000/upload`

### CODE REVIEW

File: test/unit/providers/index.test.ts [hunk 1]
```diff
@@ -1,27 +1,48 @@
...
   describe.each([
-    { name: 'fleek', set: setFleek, cid: cid.v0 },
-    { name: 'infura', set: setInfura, cid: cid.v0 },
-    { name: 'web3storage', set: setWeb3Storage, cid: cid.v1 }
-  ])('$name', ({ name, set, cid }) => {
-    it('should upload a file', async () => {
-      const result = await set(input);
+    { name: 'fleek', set: setFleek, idVersion: 'v0' },
+    { name: 'infura', set: setInfura, idVersion: 'v0' },
+    { name: 'pinata', set: setPinata, idVersion: 'v0' },
```
Comment (Sekhmet) [inline test/unit/providers/index.test.ts:30]: @bonustrack you mentioned we want Pinata support for image uploads, but we also removed Pinata support for JSON files before. Do we want to use Pinata here or something else?

Tests are failing with provided key for Pinata on CI due to "Email Verification is required before pinning content to Pinata".

File: src/upload.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const router = express.Router();
+const upload = multer({ dest: 'uploads/', limits: { fileSize: MAX_INPUT_SIZE } });
+
+router.post('/upload', upload.single('file'), async (req, res) => {
+  if (!req.file) return rpcError(res, 500, 'no file', null);
+
+  try {
+    const buffer = await fs.createReadStream(req.file.path).pipe(transformer).toBuffer();
+    const result = await Promise.any([setFleek(buffer), setPinata(buffer)]);
```
Comment (Sekhmet) [inline src/upload.ts:25]: @bonustrack you mentioned we want Pinata support for image uploads, but we also removed Pinata support for JSON files before. Do we want to use Pinata here or something else?

Comment (bonustrack) [reply src/upload.ts:25]: I've got new account on Pinata and setup different keys test passing now, we can have all of the providers for JSON, for image upload I think 3 is enough we can keep Fleek Pinata and Infura

**Review by bonustrack — COMMENTED** [review]
I've tested with Snapshot UI in local and for some reason when I try to upload a file for a second time the Pineapple server crash with this error
image

**Comment by bonustrack** [comment]
You may need to add a .gitkeep file on upload folder otherwise it wont be in the repo

**Comment by Sekhmet** [comment]
@bonustrack do we want it in the repo? I don't think we need it. `multer` will create it on startup.

**Comment by bonustrack** [comment]
@Sekhmet If thats the case then we don't really need it yea

File: .gitignore [hunk 3]
```diff
@@ -4,6 +4,7 @@ dist
 build
 .env
 coverage
+upload
```
Comment (bonustrack) [inline .gitignore:7]: The folder name should be "upload**s**" here
```suggestion
uploads
```

**Review by bonustrack — APPROVED** [review]
Works well now for me!

**Pull request merged**